matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
I can find a number of places where people refer to dense memory allocation when they refer to contiguous chunks:
[here](https://books.google.com/books?id=DqX-aWwIEJsC&pg=PA411&lpg=PA411&dq=dense+memory+allocation&source=bl&ots=a4r5UlBW9u&sig=46mXpruUj7pY4F5_HxoaR9dzvAI&hl=en&sa=X&sqi=2&ved=0ahUKEwj2q8O7sufMAhXDSiYKHf0cBx8Q6AEIKzAD#v=onepage&q=dense%20memory%20allocation&f=false) and [here](https://wiki.dlang.org/Dense_multidimensional_arrays#Dense_dynamic_arrays) seem relevant to the matrix discussion.  I don't think I am the only one who may conflates these as I have heard it used as a term of art, though personally, mostly in database discussions.

Also, I do think that the Python array is a dense/contiguous data structure. Also, if I am not mistaken, slots, strings, bytes, and structs are also all dense/contiguous in their storage. 

Defining the terms and then using them clearly is all that I desired out of my original comment.